services/core/shared/core/delegates.js

```diff
@@ -0,0 +1,126 @@
+	sortObj[sortProp] = sortOrder;
+
+	return sortObj;
+};
+
+const getSelector = (params) => {
+	const result = {};
+	result.sort = [];
+
+	if (params.limit) result.limit = params.limit;
+	if (Number(params.offset) >= 0) result.skip = params.offset;
+
+	const selector = {};
+	if (params.address) selector['account.address'] = params.address;
+	if (params.publicKey) selector['account.publicKey'] = params.publicKey;
+	if (params.secondPublicKey) selector['account.secondPublicKey'] = params.secondPublicKey;
+	if (params.username) selector.username = params.username;
+	result.selector = selector;
+
+	if (Object.getOwnPropertyNames(result.selector).length === 0) {
+		if (params.sort) result.sort.push(formatSortString(params.sort));
+		else result.sort.push({ rank: 'asc' });
+	} else if (result.sort.length === 0) delete result.sort;
+
+	return result;
+};
+
+const getDelegates = async params => {
+	const db = await pouchdb(config.db.collections.delegates.name);
+
+	let delegates = {
+		data: [],
+		meta: {},
+	};
+
+	try {
+		if (
+			!params.address
+			&& !params.publicKey
+			&& !params.secondPublicKey
+			&& !params.username
+		) throw new Error('Cannot query the DB without indexed params. Falling back to Lisk Core');
+		else {
+			const inputData = getSelector({
+				...params,
+				limit: params.limit || 10,
+				offset: params.offset || 0,
+			});
+			const dbResult = await db.find(inputData);
+
+			if (dbResult.length > 0) delegates.data = dbResult;
+			else throw new Error('Request data from Lisk Core');
+		}
+	} catch (err) {
+		logger.debug(err.message);
+
+		delegates = await coreApi.getDelegates(params);
+		delegates.data.map(delegate => {
+			delegate.id = String(delegate.rank);
+			return delegate;
+		});
+		if (delegates.data && delegates.data.length) await db.writeBatch(delegates.data);
+	}
+
+	return delegates;
+};
+
+const loadAllDelegates = async (delegateList = []) => {
+	const limit = 100;
+	const response = await getDelegates({ limit, offset: delegateList.length });
+	delegateList = [...delegateList, ...response.data];
+
+	if (response.data.length === limit) loadAllDelegates(delegateList);
+	else logger.info(`Initialized/Updated delegates cache with ${delegateList.length} delegates.`);
+};
+
+const reload = () => {
+	loadAllDelegates();
+};
+
+const getTotalNumberOfDelegates = async (params = {}) => {
+	const db = await pouchdb(config.db.collections.delegates.name);
+
+	const allDelegates = await db.findAll();
+	const relevantDelegates = allDelegates.filter(delegate => (
+		(!params.search || delegate.username.includes(params.search))
+		&& (!params.username || delegate.username === params.username)
+		&& (!params.address || delegate.account.address === params.address)
+		&& (!params.publickey || delegate.account.publicKey === params.publickey)
+		&& (!params.secpubkey || delegate.account.secondPublicKey === params.secpubkey)
+	));
+	return relevantDelegates.length;
+};
+
+module.exports = {
+	reloadDelegateCache: reload,
```

Comment:
Instead of calling reload function, can we directly call `loadAllDelegates()`?

module.exports = {
           reloadDelegateCache: loadAllDelegates,

Reply:
Even though the `reload` method simply redirects to the `loadAllDelegates` method, for now, I prefer declaring the `reload` method explicitly instead of only mapping the `loadAllDelegates` method in the `module.exports` to keep it clean and more intuitive for anyone who refers the code.